aip/general/0162.md

```diff
@@ -25,201 +25,170 @@ to the version of an API as a whole.
 }
 ```
 
-- The resource **must** contain a `revision_id` field, which **should** be a
-  string and contain a short, automatically-generated random string. A good
-  rule of thumb is the last eight characters of a [UUID version 4 (random)](UUID4).
-  - The `revision_id` field **must** document when new revisions are created
-    (see [committing revisions](#committing-revisions) below).
-  - The `revision_id` field **should** document the format of revision IDs.
-- The resource **must** contain a `revision_create_time` field, which
-  **should** be a `google.protobuf.Timestamp` (see [AIP-142][]).
+- The resource revision **must** contain a field with a message type of the
+  parent resource, with a field name of the resource singular.
+- The resource revision **must** contain a `create_time` field (see [AIP-142][]).
+- The resource revision **may** contain a repeated field `alternate_ids`, which would
+  contain a list of resource IDs that the revision is also known by (e.g. `latest`)
 
-**Note:** A randomly generated string is preferred over other methods, such as
-an auto-incrementing integer, because there is often a need to delete or revert
-revisions, and a randomly generated string holds up better in those situations.
+### Creating Revisions
 
-### Referencing revisions
+Depending on the resource, different APIs may have different strategies for
+when to create a new revision, such as:
 
-When it is necessary to refer to a specific revision of a resource, APIs
-**must** use the following syntax: `{resource_name}@{revision_id}`. For
-example:
+- Create a new revision any time that there is a change
+- Create a new revision when something important happens
+- Create a new revision when the user specifically asks
 
-    publishers/123/books/les-miserables@c7cfa2a8
+APIs **may** use any of these strategies. If a strategy is used, they **must**
+document their revision creation strategy.
 
-**Note:** The `@` character is selected because it is the only character
-permitted by [RFC 1738 §2.2][] for special meaning within a URI scheme that is
-not already used elsewhere.
+### Resource names for revisions
 
-APIs **should** generally accept a resource reference at a particular revision
-in any place where they ordinarily accept the resource name. However, they
-**must not** accept a revision in situations that mutate the resource, and
-should error with `INVALID_ARGUMENT` if one is given.
+When referring to specific revision of a resource, the subcollection name
+**must** be named `revisions`. Resource revisions have names with the format
+`{resource_name}/revisions/{revision_id}`. For example:
 
-**Important:** APIs **must not** require a revision ID, and **must** default to
-the current revision if one is not provided, except in methods specifically
-dealing with the revision history (such as rollback) where failing to require
-it would not make sense (or lead to dangerous mistakes).
+publishers/123/books/les-miserables/revisions/c7cfa2a8
```

Comment:
I suspect this should be in backticks.

Reply:
there shouldn't be a reference to current anyway, removed.

Reply:
sorry my editor put this comment in a weird place. wrapped in a code block.